frontend/src/shared/components/StatusToolTip.tsx

```diff
@@ -1,82 +1,206 @@
-        Object.entries(JSON.parse(validationReport))
-      );
-    } catch (e) {
-      const cleanedReport = validationReport.replace(/[{}]/g, "");
-      const reportArray = cleanedReport.split(",");
-      for (const r of reportArray) {
-        const [key, value] = r.split("=").map((str: String) => str.trim());
+function parseValidationReport(validationReport: string): Map<string, string> {
+  const validationReportMap = new Map<string, string>();
+  try {
+    // Parse the validation report JSON string
+    // e.g. "{"fastQs":"missing","igoComplete":"false"}"
+    return new Map(Object.entries(JSON.parse(validationReport)));
+  } catch (e) {
+    // Parse the alternative format of the validation report data
+    // e.g. "{fastQs=missing,igoComplete=false}"
+    const keyValuePairs = validationReport.replace(/[{}]/g, "").split(",");
+    for (const keyValuePair of keyValuePairs) {
+      const [key, value] = keyValuePair.split("=").map((str) => str.trim());
+      if (key && value) {
         validationReportMap.set(key, value);
       }
     }
+    return validationReportMap;
+  }
+}
 
-    validationReportList = Array.from(
-      validationReportMap,
-      ([fieldName, report]) => ({ fieldName, report })
-    );
-
-    validationReportList.map((reportObj) => {
-      reportObj.fieldName = toSentenceCase(reportObj.fieldName);
-      reportObj.report = toSentenceCase(reportObj.report);
-      return reportObj;
-    });
-  } else {
-    validationStatus = false;
-    validationReportList = [
-      {
-        fieldName: "Data error",
-        report: "Validation status is missing from this sample",
-      },
-    ];
+export function StatusTooltip({ data }: ITooltipParams<DashboardSample>) {
+  if (!data) {
+    return null;
   }
 
-  if (!validationStatus) {
-    return (
-      <div className="tooltip-styles">
-        <p>Error report for {`${primaryId}`}</p>
-        <div className="ag-theme-alpine" style={{ height: 300, width: 550 }}>
-          <AgGridReact
-            rowData={validationReportList}
-            columnDefs={columnDefs}
-            onFirstDataRendered={(params) => {
-              params.columnApi.autoSizeColumn("fieldName");
-            }}
-          ></AgGridReact>
-        </div>
-      </div>
+  const { primaryId, validationReport, validationStatus } = data;
+  const validationDataForAgGrid: SampleStatusItem[] = [];
+
+  // Populate the validationDataForAgGrid with the validation report data
+  if (validationStatus === false) {
+    const validationReportMap = parseValidationReport(validationReport!);
+    validationDataForAgGrid.push(
+      ...Array.from(validationReportMap, ([fieldName, report]) => ({
+        ...SAMPLE_STATUS_MAP[`${fieldName} ${report}`],
+      }))
     );
-  } else {
-    return "";
+    // Handle the case where a Sample is missing a corresponding Status in the database
+  } else if (validationStatus === null) {
+    validationDataForAgGrid.push(...MISSING_SAMPLE_STATUS);
   }
-};
```

Comment:
one change I'll be making here is adding a check if a sample's category is clinical or not since clinical sample metadata nodes don't have statuses attached to them @qu8n